ops/charm.py

```diff
@@ -880,16 +1006,16 @@ class RelationMeta:
 
     VALID_SCOPES = ['global', 'container']
 
-    def __init__(self, role: RelationRole, relation_name: str, raw: dict):
-        if not isinstance(role, RelationRole):
+    def __init__(self, role: RelationRole, relation_name: str, raw: '_RelationMetaDict'):
+        if not isinstance(role, RelationRole):  # pyright: reportUnnecessaryIsInstance=false
```

Comment:
Since relationmeta is only created *within* ops itself and we do static type checking for CI, maybe this is actually unnecessary.